### PR TITLE
Support specifying LINUXBUILDDIR (fixes #12)

### LINKED ISSUE #12: Allow user to specify build directory
First thanks for creating this script and the guide!

As far as I see the buildkernel script builds the kernel in the LINUXDIR="/usr/src/linux" directory, which in case of newer devices is often located on a SSD. It would be nice to allow the user to optionally specify a builddirectory to which the contents of /usr/src/linux are then copied and in which the compilation is subsequently performed.

This would allow to specify a directory on a tmpfs and thereby to work around unnecessary stressing of the SSD.

### PR DESCRIPTION
**Pull request by leezu**

### CODE REVIEW

File: buildkernel [hunk 1]
```diff
@@ -1540,8 +1540,19 @@ ensure_efi_partition_mounted() {
...
-    cd "${LINUXDIR}"
+    if [ -z ${LINUXBUILDDIR+x} ]; then
+        LINUXBUILDDIR=$LINUXDIR
+    else
+        if [ -d "$LINUXBUILDDIR" ]; then
+            show "Proceeding - Cleaning ${LINUXBUILDDIR}..."
+            rm -r ${LINUXBUILDDIR}
+        fi
+        show "Proceeding - copying from ${LINUXDIR} to ${LINUXBUILDDIR}..."
+        cp -LR ${LINUXDIR} ${LINUXBUILDDIR}
```
Comment (sakaki-) [inline buildkernel]: `cp -LR`? Not sure that's quite what is wanted - it's going to deref any symbolic links in the kernel tree. Perhaps consider `cp -ax` instead (archive copy, [staying on one filesystem](https://unix.stackexchange.com/questions/256704/what-does-cp-x-do/256714#256714) for safety)?

Comment (leezu) [inline buildkernel]: I choose it to make sure to actually copy `LINUXDIR`, which itself is a symbolic link and not to just copy the symbolic link. Given that there are relative symbolic links within the kernel source it is indeed suboptimal.
A smarter way to copy the target of `${LINUXDIR}` and not the symbolic link itself would be to use `${LINUXDIR}/` I think.

**Comment by leezu** [comment]
Another thing is to only delete the contents of `${LINUXBUILDDIR}` if it actually contains a different kernel version than `${LINUXDIR}`. If not it should be preserved to speed up the build.
I haven't looked into that yet, but do you think this could be done based on `extract_kernel_release_name` (i.e. `make -s kernelrelease`) or are there any issues with that?

**Comment by sakaki-** [comment]
Let's get a safe "always copy" version working first and then add the only-copy-if-changed option.

**Comment by leezu** [comment]
Sure. Do you see any other issues that should be addressed? Given that your comment about `cp -ax` is addressed with c808058

**Comment by sakaki-** [comment]
No, looks good... will pull now and test. If it doesn't break anything I'll release a new version, in around a week or so. Thanks for the work!

**Comment by sakaki-** [comment]
I have reworked this slightly after testing (please see comments to commit 57da2b9).
Would you mind checking the new version works OK at your end?
If no further problems, I will then release.
Thanks, sakaki

**Comment by leezu** [comment]
Thanks, your reworked version works fine for me. I have tested it for building 4.12.3.

Another thing is that the example directory specified in the .conf (`#LINUXBUILDDIR="/var/tmp"`) by default is not a tmpfs. It could be changed to `#LINUXBUILDDIR="/tmp"` or we assume the user is smart enough to recognize this and to configure `/var/tmp/buildkernel` to be mounted as tmpfs in `/etc/fstab`.

**Comment by sakaki-** [comment]
I think we can assume the user smart enough, if they are changing config variables like this.
Released now: 1.0.24.
Thanks for taking the time to file the PR!
Best, sakaki

**Pull request merged**